### ISSUE TITLE
\def and others slow

### ISSUE THREAD

**Issue by projekter** [texstudio-org/texstudio#597]
When I try to type a definition, e.g. with `\def`, `\edef`, `\let`, ..., as soon as TeXstudio recognizes the command, typing becomes slow and CPU consumption is very high.
Examples:
`\def\thisisatestmacro{content}`
Starting with the `f` of `\def`, the described behavior begins. `f\thisisatestmacro{` is pretty slow (especially on battery without power supply, when the CPU speed is regulated down, it can easily take seconds until TeXstudio shows the characters that I already typed), `content}` then is back to normal speed.
`\let\anothermacro\something`
Starting with the `t` of `\let`, everything is slow. This can be improved by using the syntax with `=`:
`\let\anothermacro=\something`
Now after the `=`, CPU is back to normal.
Obviously, skipping one letter from `\def` or `\let` and only inserting it at the very end is a working, but very inconvenient workaround.

`\usepackage{...}` is also a very slow command when typing the package name, but this is (probably) due to calling `findtexmf` (would be great to have an option to disable this), while here no external application is involved. And typing definitions is much more often used than changing the included packages, at least when writing a package.

### Environment
- TeXstudio: 2.12.16 (present in all previous versions I remember)
- Qt: 5.12.1
- OS: Windows 10
- TeX distribution: miktex
- Any completion feature I found is disabled

**Comment by manudroid19** [comment]
this is largely tested and I don't find it normal in any way. In what hardware are you running TeXstudio??

**Comment by projekter** [comment]
Schenker Office 15 Laptop
Intel Core i7-7700T (Kaby Lake), 8 cores (four real) @ 2.90GHz (2x32 KB L1, 256 KB L2, 8 MB L3 cache), no overclocking
32 GB DDR4-2133 RAM (1066 MHz) dual channel
no other computational expensive processes in the background

**Comment by sunderme** [comment]
please provide a complete (tested) example, in case the surrounding is important

**Comment by manudroid19** [comment]
you can also try to delete C:\Users\your_user\AppData\Roaming\TeXstudio folder to clean any misconfiguration in TexStudio installation. Does this happen with every file?

**Comment by projekter** [comment]
Indeed, this happens with every file, including an empty file when I just start typing.
Deleting the folder did not change anything. However, I found that the editor options for interactive checks have a large influence on this. By unchecking "Syntax", TeXstudio is fast enough for convenient working.
To nail down the critical options, I restricted TeXstudio to one core and observed its CPU consumption for extremely fast typing.
- Interactive checks completely disabled (and I mean every checkbox unchecked - if they are all still checked and only "interactive checks" are unchecked, timing is as if they were on): up to 15%
- Only spell checks: up to 20%
- Only grammer: up to 25%
- Only citation: up to 22%
- Only cross-references: up to 20%
- Only syntax: up to 30%
- Only packages: up to 22%
- All of them: up to 35%
When I inspected the cases "none" and "all" with ProcessMonitor, I could not find any different behavior, so the slowness is not due to file or registry read/writes.
